control-base/src/main/java/fi/nls/oskari/control/feature/AbstractFeatureHandler.java

```diff
@@ -77,9 +77,9 @@ protected int getLayerId(String layerId) throws ActionParamsException {
         return id;
     }
 
-    protected String postPayload(OskariLayer layer, String payload) throws ActionException {
+    protected String postPayload(OskariLayer layer, String payload, String url) throws ActionException {
```

Comment:
Remove the OskariLayer from params as it's only needed for username/password. Add user/pass to params instead.